File: examples/elixir/basic-server/test/support/basic_server_probe.ex
defmodule BasicServerProbe do
  @moduledoc false

  @spec handle_get_type_closure(term(), Basic.GetTypeClosureInput.t(), term()) ::
          {:ok, Basic.GetTypeClosureOutput.t()} | {:error, term()}
  def handle_get_type_closure(_ctx, input, _meta) do
    {:ok,
     %Basic.GetTypeClosureOutput{
       basic_string: input.name,
       basic_boolean: input.verbose
     }}
  end
end

File: examples/erlang/basic-server/test/basic_server_probe.erl
-module(basic_server_probe).

-include("basic_types.hrl").

-export([handle_get_type_closure/3]).

%% Test double: echoes decoded input fields into a minimal output record.
handle_get_type_closure(_Ctx, #get_type_closure_input{name = Name, verbose = Verbose}, _Meta) ->
    {ok, #get_type_closure_output{
        basic_string = Name,
        basic_boolean = Verbose
    }}.
